### PR TITLE
Core & Internals: Fix Flask endpoints; #1817

### PR DESCRIPTION
**Pull request by bziemons**
Rename lock, replica blueprints, because of duplicate naming resulting in errors when adding both to the same application.
Fix Flask endpoints compared to web.py.
Add global account limit endpoint to Flask.
Add `parse_scope_name` for parsing scope and name following the schema.
Add `try_stream` for all streaming endpoints.
Fix did endpoint and add missing follow endpoint.
Fix archive endpoint.
Add missing authentication endpoints.

Caveats
-----------

OIDC authentication templates were created for web.py only. The outputs on the Flask side are simple one-liners in text instead of HTML. As those endpoints shouldn't be handled as REST APIs, this doesn't break anything.

### CODE REVIEW

**Comment by bziemons** [comment]
Can be reviewed, but I am going to have another look at the generated documentation, since I am changing a few endpoints here.

**Comment by bziemons** [comment]
ready for review & merge. I would suggest merging this as a patch, because that will prevent issues with cherry-picking to the release branch & fix documentation issues.

**Comment by bari12** [comment]
@tbeerman please have a look on this as well.

File: lib/rucio/web/rest/flaskapi/v1/archive.py [hunk 1]
```diff
@@ -42,27 +46,32 @@ class Archive(MethodView):
     """ REST APIs for archive. """
 
     @check_accept_header_wrapper_flask(['application/x-json-stream'])
-    def get(self, scope, name):
+    def get(self, scope_name):
```
Comment (tbeerman) [inline lib/rucio/web/rest/flaskapi/v1/archive.py:49]: why this change?

Comment (bziemons) [reply lib/rucio/web/rest/flaskapi/v1/archive.py:49]: because schemas can configure an arbitary `SCOPE_NAME_REGEX`[(1)] and this is backwards-compatible.

[(1)]: https://github.com/rucio/rucio/blob/4ab624fd77949d97a3498e2d31ce58bfd6eef135/lib/rucio/common/schema/cms.py#L391

Comment (tbeerman) [reply lib/rucio/web/rest/flaskapi/v1/archive.py:49]: ah right, I forgot about the schema.

Comment (bziemons) [reply lib/rucio/web/rest/flaskapi/v1/archive.py:49]: I am all open for a better solution, if you have any :D

Comment (tbeerman) [reply lib/rucio/web/rest/flaskapi/v1/archive.py:49]: not really. I just missed the step with the schema matching. all good from my side.

**Review by tbeerman — APPROVED** [review]

File: lib/rucio/web/rest/flaskapi/v1/exporter.py [hunk 2]
```diff
@@ -71,6 +74,7 @@ def get(self):
 
 export_view = Export.as_view('scope')
 bp.add_url_rule('/', view_func=export_view, methods=['get', ])
+# FIXME: Add '' rule
```
Comment (bari12) [inline lib/rucio/web/rest/flaskapi/v1/exporter.py:77]: What are these for?

**Review by bari12 — APPROVED** [review]
This looks very good. Thanks a lot @bziemons 👍 
Due to the size of it we will have to test it carefully though. Next by adding it to the tests but then also by partial integration deployment.

File: lib/rucio/web/rest/flaskapi/v1/exporter.py (hunk 2 above)
Comment (bziemons) [reply lib/rucio/web/rest/flaskapi/v1/exporter.py:77]: Because the blueprints do not have the url prefix, adding this rule would currently break the blueprint since it would add a rule for an empty url. These are basically notes for myself I followed up on #3990
